untrusted/test_enclave/src/enclave.c

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+/*
+ * A basic function that just adds 2 to its input
+ */
+void ecall_add_2(int input, int *sum) {
+    *sum = input + 2;
+}
```

Comment:
This should be in rust.

```
#[no_mangle]
pub fn ecall_add_2(input: i32, sum: &i32) {
    *sum = input + 2;
}
```
should do the same job.

Reply:
TBH I was definitely in my C comfort zone when I wrote this.
However digging into this more it runs into the issue the main repo has with the enclave binaries.  Writing the implementation in rust requires one of two approaches:
* Invoking cargo build as a separate step in the build.rs
* Invoking rustc directly and trying to get the correct flags and linking the file into the final lib.

I created https://github.com/mobilecoinfoundation/sgx/issues/4 to capture this.  I think once the trusted FFI gets created and tested we'll have a better idea of what the longer term solution may look like.